nncf/quantization_metric.py

```diff
@@ -0,0 +1,96 @@
+from texttable import Texttable
+
+from nncf import create_compressed_model, NNCFConfig
+from nncf.quantization.layers import SymmetricQuantizer 
+
+
+
+class BaseQuantizationMetric:
+    def __init__(self, model_name: str, qconfig: NNCFConfig, orig_model):
+        self.model_name = model_name
+        self._qconfig = qconfig
+        self._orig_model = orig_model()
+        self.compression_ctrl, self.compressed_model = create_compressed_model(self._orig_model, self._qconfig,
```

Comment:
This isn't a test feature, but rather a real production/usability grade feature. Therefore it should be more flexible than this. 

Creating an entire new model just to measure metrics corresponding to the config is not what was in mind here - rather, there should be an option to analyze the quantized model at any time during the quantized model's lifecycle where it is possible to do so.